cpp/src/arrow/util/armce-util.h

```diff
@@ -0,0 +1,106 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef ARROW_UTIL_ARMCE_UTIL_H
+#define ARROW_UTIL_ARMCE_UTIL_H
+
+#include "my_config.h"
+
+namespace arrow {
+
+#if defined(__GNUC__) && defined(__linux__) && defined(ARROW_HAVE_ARMCE)
+
+#include <sys/auxv.h>
+#include <asm/hwcap.h>
+#ifndef HWCAP_CRC32
+#define HWCAP_CRC32 (1 << 7)
+#endif
+static inline uint32_t crc32c_runtime_check(void)
+{
+        unsigned long auxv = getauxval(AT_HWCAP);
```

Comment:
Our style guide mandates 2-space indentation. I recommend you run the style checker using `make clang-format`.

Reply:
yeah, thanks. I'll fix it.